receiver/httpreceiver/logs.go

```diff
@@ -0,0 +1,232 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httpreceiver
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+	"unicode"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+)
+
+type httpLogsReceiver struct {
+	path              string
+	serverSettings    *confighttp.HTTPServerSettings
+	telemetrySettings component.TelemetrySettings
+	server            *http.Server
+	consumer          consumer.Logs
+	wg                *sync.WaitGroup
+	logger            *zap.Logger
+}
+
+// newHTTPLogsReceiver returns a newly configured httpLogsReceiver
+func newHTTPLogsReceiver(params receiver.CreateSettings, cfg *Config, consumer consumer.Logs) (*httpLogsReceiver, error) {
+	return &httpLogsReceiver{
+		path:              cfg.Path,
+		serverSettings:    cfg.ServerSettings,
+		telemetrySettings: params.TelemetrySettings,
+		consumer:          consumer,
+		wg:                &sync.WaitGroup{},
+		logger:            params.Logger,
+	}, nil
+}
+
+// Start calls startListening
+func (r *httpLogsReceiver) Start(ctx context.Context, host component.Host) error {
+	return r.startListening(ctx, host)
+}
+
+// startListening starts serve on the server using TLS depending on receiver configuration
+func (r *httpLogsReceiver) startListening(_ context.Context, host component.Host) error {
+	r.logger.Debug("starting receiver HTTP server")
+	var err error
+	r.server, err = r.serverSettings.ToServer(host, r.telemetrySettings, r)
+	if err != nil {
+		return err
```

Comment:
Can we add wrapping to all these errors? So this one would be `fmt.Errorf("to server: %w", err)`